File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.ts
function searchMatrix(matrix: number[][], target: number): boolean {

    let row = 0;
    let col = matrix[0].length - 1;

    while (row < matrix.length && col >= 0) {
        let current = matrix[row][col];
        if (current === target)
            return true;
        else if (current > target) {
            col--;
        }
        else {
            row++;
        }
    }

    return false;
};

File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.java
class Solution {
    public boolean searchMatrix(int[][] mat, int target) {
        int m = mat.length;
        int n = mat[0].length;
        for (int arr[] : mat) {
            int i = 0, j = n - 1;
            while (i <= j) {
                int mid = (i + j) / 2;
                // System.out.println(mid);
            
                if (arr[mid] > target)
                    j= mid-1;
                else if (arr[mid] < target)
                    i= mid+1;
                else {
                    return true;
                }
            }
        }
        return false;
    }
}